### PR TITLE
[virt] DPDK Checkup: Update user facing API

### PR DESCRIPTION
**Pull request by orelmisan**
Following an upstream user-facing API change, update ConfigMap related parts.

Version(s):

4.14+

Issue:

Link to docs preview:

https://62967--docspreview.netlify.app/openshift-enterprise/latest/virt/monitoring/virt-running-cluster-checkups.html#virt-checking-cluster-dpdk-readiness_virt-running-cluster-checkups

QE review:
- [X] QE has approved this change.

Additional information:

~~Depends on PR #62804, please review only the last commit.~~

### CODE REVIEW

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 1]
```diff
@@ -6,36 +6,25 @@
...
-[IMPORTANT]
-====
-The traffic generator pod created by the checkup has elevated privileges:
-
-* It runs as root.
-* It has a bind mount to the node's file system.
-
-The container image of the traffic generator is pulled from the upstream Project Quay container registry.
-====
+* The cluster is configured to run DPDK applications.
```
Comment (yossisegev) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:22]: Duplicated line.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 2]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Indicates error receiving packets to traffic generator
+
+|`status.result.trafficGenActualNodeName`
+|Node name on which the traffic generator VM was scheduled
+
+|`status.result.vmUnderTestActualNodeName`
+|Node name on which the VM under test was scheduled
+
+|`status.result.vmUnderTestReceivedPackets`
+|Number of packets packets received on the VM under test
```
Comment (yossisegev) [inline modules/virt-dpdk-config-map-parameters.adoc:93]: "packets" duplicated

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: modules/virt-dpdk-config-map-parameters.adoc (hunk 2 above)
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Good catch!
Done.

File: modules/virt-checking-cluster-dpdk-readiness.adoc (hunk 1 above)
Comment (orelmisan) [reply modules/virt-checking-cluster-dpdk-readiness.adoc:22]: This change is from PR #62804 that this PR depends on.
The lines are not duplicated (one line talks about the cluster and another on the project).

Comment (yossisegev) [reply modules/virt-checking-cluster-dpdk-readiness.adoc:22]: You are correct.
Sorry.

**Comment by orelmisan** [comment]
[Change](https://github.com/openshift/openshift-docs/compare/6f968b1881ad5d9ebb8a5b92e856d9c0f4bfd78b..8df6c72bd7332cdcaaed93d47b9cc85f259a0f6a): Answered a review [comment](https://github.com/openshift/openshift-docs/pull/62967#discussion_r1279657982).

**Comment by yossisegev** [comment]
/lgtm
(CNV network QE)

**Review by RamLavi — APPROVED** [review]
/lgtm

**Comment by yossisegev** [comment]
/lgtm
(CNV network QE)

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 3]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Checkup start timestamp
+
+|`status.completionTimestamp`
+|Checkup completion timestamp
+
+|`status.result.trafficGenSentPackets`
+|Number of packets sent from the traffic generator
+
+|`status.result.trafficGenOutputErrorPackets`
+|Indicates error sending packets from traffic generator
```
Comment (sjhala-ccs) [inline modules/virt-dpdk-config-map-parameters.adoc:81]:
```suggestion
|The number of error packets sent from the traffic generator.
```

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 4]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+
+|`status.result.trafficGenInputErrorPackets`
+|Indicates error receiving packets to traffic generator
```
Comment (sjhala-ccs) [inline modules/virt-dpdk-config-map-parameters.adoc:84]:
```suggestion
|The number of error packets received by the traffic generator.
```

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 5]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Node name on which the traffic generator VM was scheduled
+
+|`status.result.vmUnderTestActualNodeName`
+|Node name on which the VM under test was scheduled
+
+|`status.result.vmUnderTestReceivedPackets`
+|Number of packets received on the VM under test
+
+|`status.result.vmUnderTestRxDroppedPackets`
+|Indicates ingress packets that were dropped from DPDK application
```
Comment (sjhala-ccs) [inline modules/virt-dpdk-config-map-parameters.adoc:96]:
```suggestion
|The ingress traffic packets that were dropped by the DPDK application.
```

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 6]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+
+|`status.result.vmUnderTestTxDroppedPackets`
+|Indicates egress packets were dropped from the DPDK application
```
Comment (sjhala-ccs) [inline modules/virt-dpdk-config-map-parameters.adoc:99]:
```suggestion
|The egress traffic packets that were dropped from the DPDK application.
```

**Review by sjhala-ccs — COMMENTED** [review]
Some formatting and wording suggestions, let me know if you have any questions.

**Comment by orelmisan** [comment]
[Change](https://github.com/openshift/openshift-docs/compare/118d9fdd1f64e120220f3aa3287c7f1516b15562..9f10af7576a42284e031240767eea6e4e974d72f): Answered review [comments](https://github.com/openshift/openshift-docs/pull/62967#pullrequestreview-1559195873).

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 7]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
 |====
+
+.DPDK checkup config map output fields
+[cols="1,1", options="header"]
+|====
+|Parameter
+|Description
+
+|`status.succeeded`
+|Has the checkup succeeded
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 8]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+
+|`status.failureReason`
+|Failure reason in case of a failure
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 9]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+
+|`status.startTimestamp`
+|Checkup start timestamp
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 10]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+
+|`status.completionTimestamp`
+|Checkup completion timestamp
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 11]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|`spec.param.portBandwidthGbps`
+|The maximum bandwidth of the SR-IOV NIC. The default value is 10Gbps.
 |False
 
 |`spec.param.verbose`
 |When set to `true`, it increases the verbosity of the checkup log. The default value is `false`.
 |False
 |====
+
+.DPDK checkup config map output fields
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 12]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Indicates error receiving packets to traffic generator
+
+|`status.result.trafficGenActualNodeName`
+|Node name on which the traffic generator VM was scheduled
+
+|`status.result.vmUnderTestActualNodeName`
+|Node name on which the VM under test was scheduled
+
+|`status.result.vmUnderTestReceivedPackets`
+|Number of packets received on the VM under test
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 13]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Indicates error sending packets from traffic generator
+
+|`status.result.trafficGenInputErrorPackets`
+|Indicates error receiving packets to traffic generator
+
+|`status.result.trafficGenActualNodeName`
+|Node name on which the traffic generator VM was scheduled
+
+|`status.result.vmUnderTestActualNodeName`
+|Node name on which the VM under test was scheduled
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 14]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Number of packets sent from the traffic generator
+
+|`status.result.trafficGenOutputErrorPackets`
+|Indicates error sending packets from traffic generator
+
+|`status.result.trafficGenInputErrorPackets`
+|Indicates error receiving packets to traffic generator
+
+|`status.result.trafficGenActualNodeName`
+|Node name on which the traffic generator VM was scheduled
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc (hunk 4 above)
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc (hunk 5 above)
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc (hunk 6 above)
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc (hunk 3 above)
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

File: modules/virt-dpdk-config-map-parameters.adoc [hunk 15]
```diff
@@ -23,55 +23,78 @@ The following table shows the mandatory and optional parameters that you can set
...
+|Failure reason in case of a failure
+
+|`status.startTimestamp`
+|Checkup start timestamp
+
+|`status.completionTimestamp`
+|Checkup completion timestamp
+
+|`status.result.trafficGenSentPackets`
+|Number of packets sent from the traffic generator
```
[the remark this answers is not in the document]
Comment (orelmisan) [reply modules/virt-dpdk-config-map-parameters.adoc]: Done.

**Comment by orelmisan** [comment]
[Change](https://github.com/openshift/openshift-docs/compare/210aaea9cf62e53bdae6d3ad24c55656caea4b7b..6caecfdb30a705df82bbe8c073fa24c0fc862bb3): Fixed page alignment

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 16]
```diff
@@ -186,18 +186,35 @@ kind: ConfigMap
...
+  status.result.trafficGenActualNodeName: worker-dpdk1 <8>
+  status.result.vmUnderTestActualNodeName: worker-dpdk2 <9>
+  status.result.vmUnderTestReceivedPackets: "480000000" <10>
+  status.result.vmUnderTestRxDroppedPackets: "0" <11>
+  status.result.vmUnderTestTxDroppedPackets: "0" <12>
+----
+<1> Specifies if the checkup is successful (`true`) or not (`false`).
+<2> The reason for failure if the checkup fails.
+<3> The time when the checkup started, in RFC 3339 time format.
+<4> The time when the checkup is complete, in RFC 3339 time format.
```
Comment (RamLavi) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:209]: `startTimestamp` uses past tense "started" while - `completionTimestamp` uses present tense "is complete". could you align `completionTimestamp` to be past, as the other params?

```suggestion
<4> The time when the checkup has completed, in RFC 3339 time format.
```

Comment (orelmisan) [reply modules/virt-checking-cluster-dpdk-readiness.adoc]: Done.

**Comment by orelmisan** [comment]
[Change](https://github.com/openshift/openshift-docs/compare/6caecfdb30a705df82bbe8c073fa24c0fc862bb3..c6c43c190c225c899ea1dba4a6686c6ec557499a): Answered review [comment](https://github.com/openshift/openshift-docs/pull/62967#discussion_r1283104076).

**Review by RamLavi — APPROVED** [review]
/lgtm

**Review by sjhala-ccs — COMMENTED** [review]
LGTM

**Pull request merged**